File: api/src/types/attribute.rs
#[derive(Debug, Clone, Copy, PartialEq, sqlx::Type, serde::Serialize, serde::Deserialize)]
#[sqlx(type_name = "attribute", rename_all = "lowercase")]
#[serde(rename_all = "lowercase")]
pub enum Attribute {
    Strength,
    Dexterity,
    Constitution,
    Intelligence,
    Wisdom,
    Charisma,
}

File: view/src/lib/types/Attribute.ts
export enum Attribute {
  Strength = "strength",
  Dexterity = "dexterity",
  Constitution = "constitution",
  Intelligence = "intelligence",
  Wisdom = "wisdom",
  Charisma = "charisma",
}

export const DisplayAttribute: Readonly<Record<Attribute, string>> = {
  [Attribute.Strength]: "Strength",
  [Attribute.Dexterity]: "Dexterity",
  [Attribute.Constitution]: "Constitution",
  [Attribute.Intelligence]: "Intelligence",
  [Attribute.Wisdom]: "Wisdom",
  [Attribute.Charisma]: "Charisma",
} as const;

export const Attributes: Attribute[] = Object.values(Attribute);
